components/motor/gpio/basic.go

```diff
@@ -186,39 +183,51 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 		errs = multierr.Combine(errs, m.EnablePinHigh.Set(ctx, true, extra))
 	}
 
+	m.powerPct = powerPct
+	powerPct = math.Min(powerPct, m.maxPowerPct)
+	powerPct = math.Max(powerPct, -1*m.maxPowerPct)
+
+	// pin calls on builtin baords will only take poisitve power, so the absolute value is required to be passed
+	// to pwm pins.
+	pwmPower := math.Abs(powerPct)
+
 	var pwmPin board.GPIOPin
 
 	switch m.motorType {
 	case ABPwm, DirectionPwm:
-		if math.Abs(powerPct) <= 0.001 {
+		if pwmPower <= 0.001 {
 			return m.turnOff(ctx, extra)
 		}
 		pwmPin = m.PWM
 	case AB:
 		switch {
+		// we don't have a sense of directionality passed into this function for AB pin drivers
+		// so we do the check on the power pct, use the pwmPower to calculate the duty cycle for the
+		// A and B pins connected to the motor driver.
+		// Note: it might be better to migrate the state (on, powerPct)to the parent SetPower call
+		// and have this function handle pin value setting, - ie migrate the pin logic from the parent
+		// SetPower to this function and rename it setPins
 		case powerPct >= 0.001:
 			pwmPin = m.B
 			if m.dirFlip {
 				pwmPin = m.A
 			}
-			powerPct = 1.0 - math.Abs(powerPct) // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
+			pwmPower = 1.0 - pwmPower // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
 		case powerPct <= -0.001:
 			pwmPin = m.A
 			if m.dirFlip {
 				pwmPin = m.B
 			}
-			powerPct = 1.0 - math.Abs(powerPct) // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
+			pwmPower = 1.0 - pwmPower // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
```

Comment:
this the abs needs to be put back here, otherwise we never enter this case

Reply:
the pwmPower is already absolute from powerPct. powerPct is untouched when passed into the function

Reply:
i think i just confused myself, the cases all use powerPct still on my local branch

Reply:
and i got confused by the multiple fields containing power being used